### ISSUE TITLE
truenas_share_nfs issue w/ TrueNas scale

### ISSUE THREAD

**Issue by maurizio-vitale** [dariusbakunas/terraform-provider-truenas#9]
On TrueNAS-SCALE-22.12.0 the NFS share endpoint has `path` parameter but no `paths` parameter.

Just a data point, not sure what would be the best fix to maintain compatibility with other users. For me, I'll probably hardcode something that works, in the end I only need datasets and NFS shares (and datasets work just fine)

**Comment by dariusbakunas** [comment]
Right, this is 2nd api issue already and I think this is not the end.. with cloud providers its easy, there is only one - latest api and it is pretty consistent, this is different. I may need to create multiple versions of TrueNAS go SDK and ideally keep the same resource schema for all TrueNAS versions if that is even possible. It probably makes sense to have explicit provider setting to select specific TrueNAS version instead of trying to determine it automatically.

**Comment by songhanpoo** [comment]
Yeah, i had the same issue.

**Comment by imyller** [comment]
Any progress on this one? 

NFS sharing API has slight differences in TrueNAS CORE and SCALE.

https://www.truenas.com/docs/api/core_rest_api.html
vs
https://www.truenas.com/docs/api/scale_rest_api.html

I do not think you'd have to have more variations than those two.

**Comment by lostapathy** [comment]
Just ran into this as well.  Is this something you're interested in getting going, @dariusbakunas ?  Any suggestions on how you'd like to see it tackled if somebody takes a run at it?

**Comment by imyller** [comment]
The breaking change in TrueNAS SCALE API for NFS shares was made in 22.12.2 -> 22.12.3 minor release.

Also on 2023-10-24 TrueNAS released latest stable TrueNAS SCALE 23.10 Cobia version which continues to use the same updated API. 

For this reason the `truenas-go-sdk` library fails to modify NFS shares with SCALE version 22.12.3 or newer. Most common reason for failures is the introduced difference between `path` (single string) vs. `paths` (array of strings) field in NFS share API.

I think `truenas-go-sdk` should:
a) allow field `path` or `paths` in their correct formats, but not both. Leaving responsibility for the user to choose correct API format.
or
b) support two endpoints for pre-and-post 22.12.3 API version (with or without automatic version detection)
